lib/backend/dynamo/dynamodbbk.go

```diff
@@ -548,63 +548,115 @@ func (b *Backend) Update(ctx context.Context, item backend.Item) (*backend.Lease
 	}
-	return &backend.GetResult{Items: values}, nil
-}
 
-func (b *Backend) getAllRecords(ctx context.Context, startKey, endKey backend.Key, limit int) (*getResult, error) {
-	var result getResult
+	return func(yield func(backend.Item, error) bool) {
+		count := 0
+		defer func() {
+			if count == backend.DefaultRangeLimit {
+				b.logger.WarnContext(ctx, "Range query hit backend limit. (this is a bug!)", "start_key", params.StartKey, "limit", backend.DefaultRangeLimit)
+			}
+		}()
+
+		paginator := dynamodb.NewQueryPaginator(b.svc, &input)
+		for paginator.HasMorePages() {
+			page, err := paginator.NextPage(ctx)
+			if err != nil {
+				yield(backend.Item{}, convertError(err))
+				return
+			}
 
-	// this code is being extra careful here not to introduce endless loop
-	// by some unfortunate series of events
-	for i := 0; i < backend.DefaultRangeLimit/100; i++ {
-		re, err := b.getRecords(ctx, prependPrefix(startKey), prependPrefix(endKey), limit, result.lastEvaluatedKey)
-		if err != nil {
-			return nil, trace.Wrap(err)
-		}
-		result.records = append(result.records, re.records...)
-		// If the limit was exceeded or there are no more records to fetch return the current result
-		// otherwise updated lastEvaluatedKey and proceed with obtaining new records.
-		if (limit != 0 && len(result.records) >= limit) || len(re.lastEvaluatedKey) == 0 {
-			if len(result.records) == backend.DefaultRangeLimit {
-				b.logger.WarnContext(ctx, "Range query hit backend limit. (this is a bug!)", "start_key", startKey, "limit", backend.DefaultRangeLimit)
+			for _, itemAttributes := range page.Items {
+				var r record
+				if err := attributevalue.UnmarshalMap(itemAttributes, &r); err != nil {
+					yield(backend.Item{}, convertError(err))
+					return
+				}
+
+				item := backend.Item{
+					Key:      trimPrefix(r.FullPath),
+					Value:    r.Value,
+					Revision: r.Revision,
+				}
+				if r.Expires != nil {
+					item.Expires = time.Unix(*r.Expires, 0).UTC()
+				}
+				if item.Revision == "" {
+					item.Revision = backend.BlankRevision
+				}
+
+				if !yield(item, nil) {
+					return
+				}
+				count++
+				if params.Limit != backend.NoLimit && count >= params.Limit {
+					return
+				}
 			}
-			result.lastEvaluatedKey = nil
-			return &result, nil
 		}
-		result.lastEvaluatedKey = re.lastEvaluatedKey
 	}
-	return nil, trace.BadParameter("backend entered endless loop")
+}
+
+// GetRange returns range of elements
+func (b *Backend) GetRange(ctx context.Context, startKey, endKey backend.Key, limit int) (*backend.GetResult, error) {
+	if startKey.IsZero() {
+		return nil, trace.BadParameter("missing parameter startKey")
+	}
+	if endKey.IsZero() {
+		return nil, trace.BadParameter("missing parameter endKey")
+	}
+	if limit <= 0 {
+		limit = backend.DefaultRangeLimit
+	}
+
+	var result backend.GetResult
+	for i := range b.Items(ctx, backend.IterateParams{StartKey: startKey, EndKey: endKey, Limit: limit}) {
```

Comment:
There was a previous accepted change from edoardo that fixed this, but I'm thinking it got lost in a merge somewhere...  The error from `Items` is being ignored.

I hope some kind of lint gets added for this, because being able to silently ignore error values from an iterator seems like a serious weakness in go's iterator design.